### PR TITLE
Update adapter guidelines

### PR DESCRIPTION
**Pull request by MakisH**
Related to #25 and #19.

### CODE REVIEW

**Comment by uekerman** [comment]
Previously we had in the application case guidelines:

> - [ ] Any vertices defined through `setMeshVertex` or `setMeshVertices` need to be real locations in space and the data written to the mesh needs to be data located there, i.e., it is not allowed to decode any other information.

Thinking about this again, I think it better fits here.

File: guidelines/guidelines-adapters.md [hunk 1]
```diff
@@ -60,46 +64,42 @@ We consider an adapter fulfilling all of these criteria as *Findable* and *Acces
...
-- [ ] G.6: The repository is ready to be integrated into the [preCICE system tests](https://precice.org/dev-docs-system-tests.html): A simulation can start using an unsupervised script and there is enough information to create entries under `component-templates/` and `components.yaml`.
-- [ ] G.7: There is documentation on how to extend the adapter (i.e., documentation about the software architecture).
-- [ ] G.8: The adapter is packaged either on the expected repositories of the respective solver community, or on [Spack](https://spack.io/) ([Spack packages](https://packages.spack.io/)).
+- [ ] A.7: The logging is configurable, with levels at least "no logging, release logging, and debug logging".
+- [ ] A.8: There is a peer-reviewed paper (at least in pre-print state) with a validation study, with all data available.
+- [ ] A.9: There are contribution guidelines described in or linked from a `CONTRIBUTING.md` file.
+- [ ] A.10: There is a pull request template.
+- [ ] A.11: The configuration follows a formally defined schema, which [will be available in the future](https://github.com/precice/preeco-orga/issues/18).
```
[the remark this answers is not in the document]
Comment (MakisH) [reply guidelines/guidelines-adapters.md]: Moved the required and removed the `The adapter configuration covers the mesh name, data names, participant name, and name of the preCICE configuration file.` and `In the adapter configuration, [standard names](community-contribute-to-precice.html#naming-conventions) are possible for the coupled data (e.g., `Displacement`).`.

File: guidelines/guidelines-adapters.md [hunk 2]
```diff
@@ -35,14 +37,16 @@ Applying for research funding? Mention in your proposal which level you are plan
...
+### Required
 
-- [ ] B.1: The adapter is accompanied by at least one application case to test it, covering at least a defined minimum functionality. The test case can be provided as an independent application case or included in the adapter.
-- [ ] B.2: There is a `README.md` file with at least the following information (or links to related resources):
+We consider an adapter fulfilling all of these criteria as conforming to the preCICE standards. This means that the adapter is working, documented, plays well with other adapters from the community, and feels part of the preCICE ecosystem. Others can find, build, and run the code, and they are able to understand what is does. The preCICE maintainers are, if necessary, able to maintain the adapter (e.g. update it to newer preCICE versions).
+
+- [ ] R.1: The adapter is accompanied by one or more application cases to test it, covering an extensive part of the claimed functionality. These cases must be provided as independent application cases.
```
[the remark this answers is not in the document]
Comment (MakisH) [reply guidelines/guidelines-adapters.md]: Removed the "independent application case" point.
Do we want to enforce the application case guidelines for this case?

[the remark this answers is not in the document]
Comment (uekerman) [reply guidelines/guidelines-adapters.md]: Good point. Yes!

File: guidelines/guidelines-adapters.md [hunk 3]
```diff
@@ -60,46 +64,41 @@ We consider an adapter fulfilling all of these criteria as *Findable* and *Acces
...
+- [ ] A.5: The logging is configurable, with levels at least "no logging, release logging, and debug logging".
+- [ ] A.6: There is a peer-reviewed paper (at least in pre-print state) with a validation study, with all data available.
+- [ ] A.7: There are contribution guidelines described in or linked from a `CONTRIBUTING.md` file.
+- [ ] A.8: There is a pull request template.
+- [ ] A.9: There are unit tests to test components of the adapter, without running another simulation participant (and ideally, using the upcoming [general mocked interface](https://github.com/precice/preeco-orga/issues/4)).
+- [ ] A.10: The repository is ready to be integrated into the [preCICE system tests](https://precice.org/dev-docs-system-tests.html): A simulation can start using an unsupervised script and there is enough information to create entries under `component-templates/` and `components.yaml`.
+- [ ] A.11: There is documentation on how to extend the adapter (i.e., documentation about the software architecture).
```
[the remark this answers is not in the document]
Comment (MakisH) [reply guidelines/guidelines-adapters.md]: These are different: the one is about contributing in general (e.g., which workflows to follow) and the other is specifically about extending (i.e., architecture).

File: guidelines/guidelines-adapters.md [hunk 4]
```diff
@@ -60,46 +64,41 @@ We consider an adapter fulfilling all of these criteria as *Findable* and *Acces
...
-- [ ] G.5: There are unit tests to test components of the adapter, without running another simulation participant (and ideally, using the upcoming [general mocked interface](https://github.com/precice/preeco-orga/issues/4)).
-- [ ] G.6: The repository is ready to be integrated into the [preCICE system tests](https://precice.org/dev-docs-system-tests.html): A simulation can start using an unsupervised script and there is enough information to create entries under `component-templates/` and `components.yaml`.
-- [ ] G.7: There is documentation on how to extend the adapter (i.e., documentation about the software architecture).
-- [ ] G.8: The adapter is packaged either on the expected repositories of the respective solver community, or on [Spack](https://spack.io/) ([Spack packages](https://packages.spack.io/)).
+- [ ] A.5: The logging is configurable, with levels at least "no logging, release logging, and debug logging".
+- [ ] A.6: There is a peer-reviewed paper (at least in pre-print state) with a validation study, with all data available.
```
[the remark this answers is not in the document]
Comment (MakisH) [reply guidelines/guidelines-adapters.md]: Dropped A3.

File: guidelines/guidelines-adapters.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (MakisH) [reply guidelines/guidelines-adapters.md]: Added the requirement.

File: guidelines/guidelines-adapters.md [hunk 5]
```diff
@@ -60,46 +64,41 @@ We consider an adapter fulfilling all of these criteria as *Findable* and *Acces
   - [ ] restrictions on solver features
     - e.g., cannot be run in parallel or cannot use adaptive time stepping or cannot use higher-order elements
   - [ ] how to cite
```
[the remark this answers is not in the document]
Comment (MakisH) [reply guidelines/guidelines-adapters.md]: Removed.

**Review by uekerman — APPROVED** [review]

**Pull request merged**